sysdump/sysdump.go

```diff
@@ -0,0 +1,825 @@
+// Copyright 2021 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sysdump
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/cilium/cilium-cli/internal/utils"
+	"github.com/hashicorp/go-multierror"
+	archiver "github.com/mholt/archiver/v3"
+	log "github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+)
+
+type SysdumpOptions struct {
+	CiliumLabelSelector         string
+	CiliumNamespace             string
+	CiliumOperatorLabelSelector string
+	CiliumOperatorNamespace     string
+	HubbleLabelSelector         string
+	HubbleNamespace             string
+	HubbleRelayLabelSelector    string
+	HubbleRelayNamespace        string
+	HubbleUiLabelSelector       string
+	HubbleUiNamespace           string
+	LargeSysdumpAbortTimeout    time.Duration
+	LargeSysdumpThreshold       int32
+	Logger                      *log.Entry
+	LogsLimitBytes              int64
+	LogsSinceTime               time.Duration
+	NodeFilter                  string
+	OutputFileName              string
+	Quick                       bool
+	RESTClientGetter            genericclioptions.RESTClientGetter
+}
+
+type Sysdump interface {
+	Collect(context.Context) error
+}
+
+type sysdump struct {
+	k KubernetesClient
+	o SysdumpOptions
```

Comment:
I think that one letter variables are fine for scoped contexts, not so much as a struct member.